packages/vscode-extension/src/webviews/connectionPanel.ts

```diff
@@ -0,0 +1,251 @@
+          case 'onSaveConnection': {
+            await commands.executeCommand(
+              constants.COMMANDS.SAVE_CONNECTION_COMMAND,
+              message.connection,
+              message.password,
+            );
+            this.dispose();
+            break;
+          }
+          case 'onValidationError': {
+            void window.showErrorMessage(
+              constants.MESSAGES.CONNECTION_VALIDATION_MESSAGE,
+            );
+            break;
+          }
+        }
+      },
+      null,
+      this._disposables,
+    );
+  }
+
+  private getHtmlForWebview(webview: Webview): string {
+    const resetCssPath = Uri.joinPath(
+      this._extensionUri,
+      'resources',
+      'styles',
+      'reset.css',
+    );
+    const vscodeCssPath = Uri.joinPath(
+      this._extensionUri,
+      'resources',
+      'styles',
+      'vscode.css',
+    );
+    const connectionPanelCssPath = Uri.joinPath(
+      this._extensionUri,
+      'resources',
+      'styles',
+      'connectionPanel.css',
+    );
+
+    const connectionPanelJsPath = Uri.joinPath(
+      this._extensionUri,
+      'dist',
+      'webviews',
+      'connectionPanelController.js',
+    );
+
+    const resetCssUri = webview.asWebviewUri(resetCssPath);
+    const vscodeCssUri = webview.asWebviewUri(vscodeCssPath);
+    const connectionPanelCssUri = webview.asWebviewUri(connectionPanelCssPath);
+    const connectionPanelJsUri = webview.asWebviewUri(connectionPanelJsPath);
+    const nonce = getNonce();
+
+    // Connect by default if this is a new connection
+    const connect = this._connection?.connect ?? true;
+
+    return `
+        <!DOCTYPE html>
+        <html lang="en">
+          <head>
+            <meta charset="UTF-8">
+            <!--
+            Use a content security policy to only allow loading images from https or from our extension directory,
+            and only allow scripts that have a specific nonce.
+            -->
+            <meta http-equiv="Content-Security-Policy" content="img-src https: data:; style-src 'unsafe-inline' ${
+              webview.cspSource
+            }; script-src 'nonce-${nonce}';">
+            <meta name="viewport" content="width=device-width, initial-scale=1.0">
+            <link href="${resetCssUri.toString()}" rel="stylesheet">
+            <link href="${vscodeCssUri.toString()}" rel="stylesheet">
+            <link href="${connectionPanelCssUri.toString()}" rel="stylesheet">
+            <script nonce="${nonce}">
+              const vscode = acquireVsCodeApi();
+            </script>
+          </head>
+          <body>
+            <div class="container">
+              <div>
+                <h1>${
+                  this._connection ? 'Edit Connection' : 'Add New Connection'
+                }</h1>
+                <form class="form" action="" novalidate method="post">
+                  <input type="hidden" id="key" value="${
+                    this._connection?.key ?? getNonce(16)
```

Comment:
reuse our nonce function to generate a unique key

Reply:
nice